business/iter8.go

```diff
@@ -145,6 +148,37 @@ func (in *Iter8Service) CreateIter8Experiment(namespace string, body []byte) (mo
 	return iter8ExperimentDetail, nil
 }
 
+func (in *Iter8Service) UpdateIter8Experiment(namespace string, name string, body []byte) (models.Iter8ExperimentDetail, error) {
+	var err error
+	promtimer := internalmetrics.GetGoFunctionMetric("business", "Iter8Service", "CreateIter8Experiment")
+	defer promtimer.ObserveNow(&err)
+
+	iter8ExperimentDetail := models.Iter8ExperimentDetail{}
+	action := models.ExperimentAction{}
+	err = json.Unmarshal(body, &action)
+	if err != nil {
+		return iter8ExperimentDetail, err
+	}
+	experiment, err := in.GetIter8Experiment(namespace, name)
+	newExperimentSpec := models.Iter8ExperimentSpec{}
+	newExperimentSpec.Parse(experiment)
+	newExperimentSpec.Action = action.Action
+	var newObject []byte
+	newObject, err = json.Marshal(newExperimentSpec)
+	json, err := in.ParseJsonForCreate(newObject)
+	if err != nil {
+		return iter8ExperimentDetail, err
+	}
+
+	iter8ExperimentObject, err := in.k8s.UpdateIter8Experiment(namespace, name, string(json))
```

Comment:
The kubernetes/UpdateIter8Experiment is a generic patch which is good.
But perhaps the business/UpdateIter8Experiment should be also more specific.
Please @jadeyliu939 correct me if I'm wrong, but basically this business/UpdateIter8Experiemt only apply a change in the action, isn't it ?

Then, perhaps at business package level could be renamed or focused for just actions ?

Also, I'm still reading the whole PR to understand the other changes, perhaps I can resolve my question later.

Reply:
Yes, currently it is only a change in the action. But I am not sure if in the future, there will be other `apply` possible. So i kind of left it generic.